### PR TITLE
Add preparatory checks to the Crystal release process

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: processes/crystal-release.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ Add an issue `Crystal release X.Y.Z` in https://github.com/crystal-lang/distribu
    * (minor) Set date on milestone
 2. [ ] Prepare the changelog entry: [`crystal:scripts/github-changelog.cr`](https://github.com/crystal-lang/crystal/blob/master/scripts/github-changelog.cr)
 3. [ ] Start preparing release notes
+   * Ensure that all merged PRs are added to the milestone (check [`is:pr is:merged sort:updated-desc no:milestone`](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+no%3Amilestone)).
```
Comment (beta-ziliani) [inline processes/crystal-release.md:12]: I'd move these to the previous step, so the changelog is properly created

File: processes/crystal-release.md [hunk 2]
```diff
@@ -19,6 +21,8 @@ Add an issue `Crystal release X.Y.Z` in https://github.com/crystal-lang/distribu
       * (minor) Update language specification
       * (minor) Update compiler manual
       * (minor) Add or update guides / tutorials?
+6. [ ] Ensure that [test-ecosystem](https://github.com/crystal-lang/test-ecosystem) functions and succeeeds on master
```
Comment (beta-ziliani) [inline processes/crystal-release.md:24]: What's the difference with the one below in Source release?

Comment (straight-shoota) [reply processes/crystal-release.md:24]: This is intended as a check you do in advance of the actual release process. To make sure that test-ecosystem works at all. Doing it in advance gives time to fix it if it's broken. That's the problem we have right now, it just doesn't work.

File: processes/crystal-release.md (hunk 1 above)
Comment (straight-shoota) [reply processes/crystal-release.md:12]: That's where I wanted it to go, actually 🙈

**Review by beta-ziliani — APPROVED** [review]

**Pull request merged**